src/EditorFeatures/Core.Wpf/InheritanceMargin/MarginGlyph/InheritanceMargin.xaml

```diff
@@ -0,0 +1,215 @@
+﻿<Button x:Class="Microsoft.CodeAnalysis.Editor.InheritanceMargin.MarginGlyph.InheritanceMargin"
+        x:ClassModifier="internal"
+        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+        xmlns:imaging="clr-namespace:Microsoft.VisualStudio.Imaging;assembly=Microsoft.VisualStudio.Imaging"
+        xmlns:vsui="clr-namespace:Microsoft.VisualStudio.PlatformUI;assembly=Microsoft.VisualStudio.Shell.15.0"
+        xmlns:platformUi="clr-namespace:Microsoft.VisualStudio.PlatformUI;assembly=Microsoft.VisualStudio.Imaging"
+        ToolTip="{Binding ToolTip}"
+        AutomationProperties.Name="{Binding AutomationName}"
+        Click="Margin_OnClick">
+
+    <Button.Resources>
+        <!-- The image shown in the menu item, mark this is non-shared to make sure each menu item has its own image -->
+        <imaging:CrispImage x:Key="NonSharedIcon" x:Shared="False" Moniker="{Binding ImageMoniker}"/>
+
+        <!-- Template copied from editor -->
+        <ControlTemplate x:Key="ContextMenuTemplate" TargetType="{x:Type ContextMenu}">
```

Comment:
For whoever review the xaml, I copied some styles and control template from the editor light bulb style (which I believe orginally it is copied from shell), remove its preview pops up in order to fit our needs.

I choose this current implementation because setting the correct theme for the context menu would require breaking into the inner grid in the ControlTemplate, setting everything from scratch, and later on, I feel this is what the light bulb has done.

I really don't like this approach, so I am open to any other idea to implement this, if you know any vsct file magic or there is any good shell library that I don't know, please tell me and I am super happy to change this part.

The needs are: 
1. The MaxHeight could be set for the context menu or scrollViewer. So that if there are a lot of item in it, the.
2. The context menu and menu item needs to be correctly themed according to VS.
3. The context menu needs to be dynamically popped up.
4. Submenu is needed.
5. It needs to be shown when clicking the margin.

Reply:
Unfortunately I think the only thing to simplify would be to have styles inherit from known resource keys: https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.shell.vsresourcekeys?view=visualstudiosdk-2019 

I don't think it covers all of our needs though, and some customization would still be necessary

Reply:
Look into the doc and I feel the only usable resource key is scroll viewer. I have add that.